### PR TITLE
Added the Object Viewport Color node.

### PR DESCRIPTION
**Pull request by 3DSinghVFX**
I have added the _Object Viewport Color_ node. Blender 2.8x has _viewport color_ option in the shader nodes, so viewport color can be utilized to set instanced objects color,
Screenshot from 2020-02-19 18-47-17

Example,
Screenshot from 2020-02-19 18-42-17

### CODE REVIEW

**Comment by OmarEmaraDev** [comment]
Can we use the vectorize code effect here? Allow Extension should be enabled for the color input.

**Comment by 3DSinghVFX** [comment]
Do you mean individual colors RGB, HSV, etc ?

**Comment by OmarEmaraDev** [comment]
I mean use `codeEffects = [VectorizedSocket.CodeEffect]` like you did before.

**Comment by 3DSinghVFX** [comment]
Okay.

File: animation_nodes/nodes/object/object_viewport_color.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    bl_label = "Object Viewport Color"
+    codeEffects = [VectorizedSocket.CodeEffect]
+
+    useObjectList: VectorizedSocket.newProperty()
+    useColorList: VectorizedSocket.newProperty()
+
+    def create(self):
+        self.newInput(VectorizedSocket("Object", "useObjectList",
+            ("Object", "object", dict(defaultDrawType = "PROPERTY_ONLY")),
+            ("Objects", "objects")))
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/object/object_viewport_color.py]: The object input should have `codeProperties = dict(allowListExtension = False)`. `allowListExtension` is the property that tells Animation Nodes to repeat this list if the other lists were larger (Similar to virtual lists). It is True by default, and we don't want objects to repeat.

File: animation_nodes/nodes/object/object_viewport_color.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+        self.newInput(VectorizedSocket("Color", ["useObjectList", "useColorList"],
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/object/object_viewport_color.py]: `useObjectList` shouldn't be here.

File: animation_nodes/nodes/object/object_viewport_color.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            ("Color", "color"), ("Colors", "colors")))
+
+        self.newOutput(VectorizedSocket("Object", "useObjectList",
+            ("Object", "object", dict(defaultDrawType = "PROPERTY_ONLY")),
+            ("Objects", "objects")))
+
+    def getExecutionCode(self, required):
+        return "newSpline = self.execute_object_color(object, color)"
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/object/object_viewport_color.py]: `newSpline` shouldn't be here. You may embed `execute_object_color` in the execution code directly.

File: animation_nodes/nodes/object/object_viewport_color.py (hunk 1 above)
Comment (3DSinghVFX) [reply animation_nodes/nodes/object/object_viewport_color.py]: Okay.

File: animation_nodes/nodes/object/object_viewport_color.py (hunk 2 above)
Comment (3DSinghVFX) [reply animation_nodes/nodes/object/object_viewport_color.py]: Okay.

File: animation_nodes/nodes/object/object_viewport_color.py (hunk 3 above)
Comment (3DSinghVFX) [reply animation_nodes/nodes/object/object_viewport_color.py]: Okay.

File: animation_nodes/nodes/object/object_viewport_color.py (hunk 1 above)
Comment (3DSinghVFX) [reply animation_nodes/nodes/object/object_viewport_color.py]: Omar, I have added this code,
```
self.newInput(VectorizedSocket("Object", "useObjectList",
            ("Object", "object", dict(defaultDrawType = "PROPERTY_ONLY")),
            ("Objects", "objects")), codeProperties = dict(allowListExtension = False))
```
When I input color list larger than the input objects, it slows down the node-tree.

Comment (OmarEmaraDev) [reply animation_nodes/nodes/object/object_viewport_color.py]: This should be:
```python
self.newInput(VectorizedSocket("Object", "useObjectList",
            ("Object", "object", dict(defaultDrawType = "PROPERTY_ONLY")),
            ("Objects", "objects"),
            codeProperties = dict(allowListExtension = False)))
```

Comment (3DSinghVFX) [reply animation_nodes/nodes/object/object_viewport_color.py]: Done.

File: animation_nodes/nodes/object/object_viewport_color.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+        self.newInput(VectorizedSocket("Color", "useColorList",
+            ("Color", "color"), ("Colors", "colors")))
+
+        self.newOutput(VectorizedSocket("Object", "useObjectList",
+            ("Object", "object", dict(defaultDrawType = "PROPERTY_ONLY")),
+            ("Objects", "objects")))
+
+    def getExecutionCode(self, required):
+        return "self.execute_object_color(object, color)"
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/object/object_viewport_color.py]: Just to clarify, I want you to inline `execute_object_color` in the execution code. Don't call the function, just put the code directly.

Comment (3DSinghVFX) [reply animation_nodes/nodes/object/object_viewport_color.py]: It is done.

File: animation_nodes/ui/node_menu.py [hunk 5]
```diff
@@ -344,6 +344,8 @@ def draw(self, context):
         insertNode(layout, "an_ArmatureInfoNode", "Armature Info")
         layout.menu("AN_MT_object_utils_menu", text = "Utils")
         layout.separator()
+        insertNode(layout, "an_ObjectViewportColorNode", "Viewport Color")
```
Comment (OmarEmaraDev) [inline animation_nodes/ui/node_menu.py]: Rename the node to `an_ObjectColorOutputNode` and move it under Data Path Output node in the menu.

Comment (3DSinghVFX) [reply animation_nodes/ui/node_menu.py]: Omar, I think _Object Viewport Color_ name is more convenient and has a similar meaning as we have in the blender,
Screenshot from 2020-02-21 15-43-30

Comment (OmarEmaraDev) [reply animation_nodes/ui/node_menu.py]: But it is not really a viewport color. It is just a color that can be used for viewport rendering.

Comment (3DSinghVFX) [reply animation_nodes/ui/node_menu.py]: Okay.

File: animation_nodes/nodes/object/object_color_output.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        self.newInput(VectorizedSocket("Object", "useObjectList",
+            ("Object", "object", dict(defaultDrawType = "PROPERTY_ONLY")),
+            ("Objects", "objects"),
+            codeProperties = dict(allowListExtension = False)))
+
+        self.newInput(VectorizedSocket("Color", "useColorList",
+            ("Color", "color"), ("Colors", "colors")))
+
+        self.newOutput(VectorizedSocket("Object", "useObjectList",
+            ("Object", "object", dict(defaultDrawType = "PROPERTY_ONLY")),
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/object/object_color_output.py]: Is `dict(defaultDrawType = "PROPERTY_ONLY")` really needed here?

File: CHANGELOG.md [hunk 7]
```diff
@@ -24,6 +24,7 @@
 - Added Set Polygon Material Index node.
 - Added Deep Copy option to the Copy Object Data node.
 - Added Decompose Text node.
+- Added Object Viewport Color node.
```
Comment (OmarEmaraDev) [inline CHANGELOG.md]: Update the name here as well.

Comment (3DSinghVFX) [reply CHANGELOG.md]: Done.

File: animation_nodes/nodes/object/object_color_output.py (hunk 6 above)
Comment (OmarEmaraDev) [reply animation_nodes/nodes/object/object_color_output.py]: Have you seen this?

Comment (3DSinghVFX) [reply animation_nodes/nodes/object/object_color_output.py]: It is in the _Object Attribute Output_ node.

**Pull request merged**